aea/runtime.py

```diff
@@ -261,165 +247,79 @@ def set_loop(self, loop: AbstractEventLoop) -> None:
 
         :param loop: event loop to use.
         """
-        super().set_loop(loop)
-        self.multiplexer.set_loop(self.loop)
-        self.main_loop.set_loop(self.loop)
-        self._async_stop_lock = asyncio.Lock()
-
-    def _start(self) -> None:
-        """
-        Start runtime synchronously.
+        BaseRuntime.set_loop(self, loop)
 
-        Set event loops for multiplexer and agent run loop.
+    async def run(self) -> None:
+        """Start multiplexeor task."""
+        terminal_state = RuntimeStates.error
+        try:
+            await self.run_runtime()
+        except StopRuntime as e:
+            self._state.set(RuntimeStates.stopping)
+            terminal_state = RuntimeStates.stopped
+            if e.reraise:
+                raise e.reraise
+        except (asyncio.CancelledError, CancelledError, KeyboardInterrupt):
+            self._state.set(RuntimeStates.stopping)
+            terminal_state = RuntimeStates.stopped
+        finally:
+            await self.stop_runtime()
+            self._state.set(terminal_state)
 
-        Start runtime asynchonously in own event loop.
+    async def stop_runtime(self) -> None:
         """
-        self.set_loop(self.loop)
+        Stop runtime coroutine.
 
-        logger.debug(f"Start runtime event loop {self.loop}: {id(self.loop)}")
-        self._task = self.loop.create_task(self.run_runtime())
+        Stop main loop.
+        Agent teardown.
```

Comment:
For consistency with the other two steps:
```suggestion
        Tear down the agent.
```

Reply:
+